x/protorev/keeper/keeper_test.go

```diff
@@ -965,8 +1012,8 @@ func (s *KeeperTestSuite) setUpTokenPairRoutes() {
 	fourPool3 := types.NewTrade(0, "test/2", "Atom")
 
 	// Two-Pool Route
-	twoPool0 := types.NewTrade(0, "test/3", types.OsmosisDenomination)
-	twoPool1 := types.NewTrade(39, types.OsmosisDenomination, "test/3")
+	twoPool0 := types.NewTrade(0, "ibc/0CD3A0285E1341859B5E86B6AB7682F023D03E97607CCC1DC95706411D866DF7", types.OsmosisDenomination)
```

Comment:
Let's open up an issue to refactor the `Posthandler` logic slightly, so that we can directly test that the user's tx doesn't revert even when the gas meter runs out. 

This test is basically assuming that's going to happen by initializing a pool with a ton of ticks and very low liquidity, such that Protorev tries to swap across it and runs out. But in theory, the gas optimization could get good enough or ProtoRev could be given more gas such that the test stops running out of gas, and we wouldn't know it's not doing the thing we think it is anymore. 

I think this requires a slight refactor of the posthandler code that would allow initializing and passing in a zero gas meter, so we'd know for sure it would run out. 

One way to do this would be to create a higher-order function that takes in a gas amount as a parameter returns a Posthandler that always initializes a gas meter with that amount. 

A simpler way would be to create a helper method that the posthandler calls to wrap the logic for creating a gas meter and doing the simulations, then we could test that function directly. 

( I'm not attached to any one approach at all. )

Reply:
Added to this tracking issue: https://github.com/osmosis-labs/osmosis/issues/5858